src/internet_identity/src/main.rs

```diff
@@ -388,31 +388,33 @@ fn update_account(
     }
 }
 
-#[update]
+#[query]
 fn get_default_account(
     anchor_number: AnchorNumber,
-    _origin: FrontendHostname,
+    origin: FrontendHostname,
 ) -> Result<AccountInfo, GetDefaultAccountError> {
     check_authorization(anchor_number)
         .map_err(|err| GetDefaultAccountError::Unauthorized(err.principal))?;
 
-    Err(GetDefaultAccountError::InternalCanisterError(
-        "Function get_default_account is not implemented yet.".to_string(),
-    ))
+    let default_account_info =
+        account_management::get_default_account_for_origin(anchor_number, origin)?;
+
+    Ok(default_account_info)
 }
 
 #[update]
 fn set_default_account(
     anchor_number: AnchorNumber,
-    _origin: FrontendHostname,
-    _account_number: Option<AccountNumber>,
+    origin: FrontendHostname,
+    account_number: Option<AccountNumber>,
 ) -> Result<AccountInfo, SetDefaultAccountError> {
     check_authorization(anchor_number)
```

Comment:
I remembered now that we have a helper `anchor_operation_with_authz_check`. Worth doing in another PR though.

The idea is that all update operations are stored then in the archive canister.

Which also means that this operation or setting an account as default must be stored there as well.

Reply:
Why isn't this used in `update_account`?

Reply:
Oh I see, it's just done directly via post_operation_bookkeeping